lib/bolt/pal.rb

```diff
@@ -143,6 +143,35 @@ def list_tasks
       end
     end
 
+    def parse_param(val, type, compiler)
```

Comment:
This kind of seems like it should be in `run_task` and `run_plan`. Do we want to make the same logic work for plans?

Reply:
Counterpoint: we'd end up doing parameter matching even when called from within a plan.

Reply:
Do we want to parse JSON when the functions are called from a plan?

Reply:
This might be our best option for now. I'd allow for the task to not be found and not do any conversion in that case. We can then say you either need to setup modulepath if you want conversion, and we can work on improving it later if we think of a better way.